### PR TITLE
Set up CodeQL scans

### PR DESCRIPTION
**Pull request by artem-smotrakov**
GitHub offers code scanning that is based on CodeQL (the same engine that is used in [LGTM.com](https://lgtm.com)). Shortly, that's a static analysis tool that is able to catch various issues including security ones. Once enabled for pull requests, it would help with catching issues earlier.

The scan may be run via [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/github/finding-security-vulnerabilities-and-errors-in-your-code/enabling-code-scanning-for-a-repository#enabling-code-scanning-using-actions) for pull requests. Or, LGTM checks may be enabled, the instructions are [here](https://lgtm.com/help/lgtm/managing-automated-code-review).

This pull request adds a GitHub workflow that runs CodeQL scans for pull requests and the main branches. There results will be available:

- in the "Security" tab for project maintainers
- in each pull request (only new alerts or fixed ones)

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: "CodeQL"
+
+on:
+  schedule:
+    - cron: '40 21 * * 2'
```
Comment (cowtowncoder) [inline .github/workflows/codeql-analysis.yml:5]: I forget how to read these expressions -- what'd be the frequence of timed runs?

Comment (cowtowncoder) [reply .github/workflows/codeql-analysis.yml:5]: As per https://crontab.cronhub.io/ apparently "At 09:40 PM, only on Tuesday", i.e. once per week. Which sounds good.

**Comment by cowtowncoder** [comment]
Merged, but now slightly worried about cpu time spent (over 5 minutes), wrt Gitlab CI resource allocation.
Seems like quite a bit; wondering if this should be cron-only instead of all check-ins and PRs.

**Comment by artem-smotrakov** [comment]
> Merged, but now slightly worried about cpu time spent (over 5 minutes), wrt Gitlab CI resource allocation.
> Seems like quite a bit; wondering if this should be cron-only instead of all check-ins and PRs.

I'd prefer to keep it running for pull requests because it would help with catching possible issues earlier. If it starts hitting the CPU limits, we can try dropping scans for 1-2 branches.

**Comment by cowtowncoder** [comment]
Sounds good; agree wrt PRs.

**Pull request merged**